modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -843,6 +963,284 @@ static void run_arithm_s(DST out[], const SRC in[], int width, int chan,
         CV_Error(cv::Error::StsBadArg, "unsupported number of channels");
 }
 
+#if CV_SIMD
+CV_ALWAYS_INLINE void absdiffc_short_store_c1c2c4(short* out_ptr, const v_int32& c1, const v_int32& c2)
+{
+    vx_store(out_ptr, v_pack(c1, c2));
+}
+
+CV_ALWAYS_INLINE void absdiffc_short_store_c1c2c4(ushort* out_ptr, const v_int32& c1, const v_int32& c2)
+{
+    vx_store(out_ptr, v_pack_u(c1, c2));
+}
+
+template<typename T>
+CV_ALWAYS_INLINE int absdiffc_simd_c1c2c4(const T in[], T out[],
+                                          const v_float32& s, const int length)
+{
+    constexpr int nlanes = static_cast<int>(v_uint16::nlanes);
+    if (length < nlanes)
+        return 0;
+
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - nlanes; x += nlanes)
+        {
+            v_float32 a1 = v_cvt_f32(vx_load_expand(in + x)),
+                a2 = v_cvt_f32(vx_load_expand(in + x + nlanes / 2));
+
+            absdiffc_short_store_c1c2c4(&out[x], v_round(v_absdiff(a1, s)),
+                v_round(v_absdiff(a2, s)));
+        }
+
+        if (x < length && (in != out))
+        {
+            x = length - nlanes;
+            continue;  // process one more time (unaligned tail)
+        }
+        break;
+    }
+    return x;
+}
+
+template<>
+CV_ALWAYS_INLINE int absdiffc_simd_c1c2c4<uchar>(const uchar in[], uchar out[],
+                                                 const v_float32& s, const int length)
+{
+    constexpr int nlanes = static_cast<int>(v_uint8::nlanes);
+
+    if (length < nlanes)
+        return 0;
+
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - nlanes; x += nlanes)
+        {
+            v_float32 a1 = v_cvt_f32(vx_load_expand_q(in + x)),
+                      a2 = v_cvt_f32(vx_load_expand_q(in + x + nlanes / 4)),
+                      a3 = v_cvt_f32(vx_load_expand_q(in + x + nlanes / 2)),
+                      a4 = v_cvt_f32(vx_load_expand_q(in + x + 3 * nlanes / 4));
```

Comment:
> vx_load_expand_q
> vx_load_expand_q
> vx_load_expand_q
> vx_load_expand_q

Reduce pressure on CPU's LOAD units. Fetched memory is equal to `vx_load(in + x)`.
Load `v_uint8` first and then repack in registers.

Reply:
I didn't quite understand your proposal. could you please clarify your idea?

Reply:
Replace 4 load instruction to one.

Reply:
But I need initialize 4 vectors for I need for further work with them. How can load four vectors with one load call?

Reply:
@terfendail Could you please comment or clarify Alexander's proposal? How will the approach that Alexander recommends affect performance?

Reply:
I think Alexander means something like
```
v_uint16 ld0, ld1;
v_expand(vx_load(in+x), ld0, ld1);
v_float32 a1 = v_cvt_f32(v_expand_low(ld0));
v_float32 a2 = v_cvt_f32(v_expand_high(ld0));
v_float32 a3 = v_cvt_f32(v_expand_low(ld1));
v_float32 a4 = v_cvt_f32(v_expand_high(ld1));
```

Reply:
@terfendail Ok. Thank you so much for clarification!

Reply:
> I think Alexander means something like
> 
> ```
> v_uint16 ld0, ld1;
> v_expand(vx_load(in+x), ld0, ld1);
> v_float32 a1 = v_cvt_f32(v_expand_low(ld0));
> v_float32 a2 = v_cvt_f32(v_expand_high(ld0));
> v_float32 a3 = v_cvt_f32(v_expand_low(ld1));
> v_float32 a4 = v_cvt_f32(v_expand_high(ld1));
> ```


@alalek I applied your proposal for 8U and gather performance report for AVX512 vectors. I observed average for 8U performance degradation equals to 12.6%. For 8UC3 test cases performance degradation is up to 33.3%. So I wouldn't like to apply this proposal to my snippet. Please take a look at the comparative performance report: //cc @dmatveev 

[vx_load_expand_q_vs_v_expand_high_low.xlsx](https://github.com/opencv/opencv/files/5876000/vx_load_expand_q_vs_v_expand_high_low.xlsx)

Reply:
Changes for the applied proposal you can see in the "Performance experiment" commit.